questionpy_server/worker/worker/base.py

```diff
@@ -225,8 +235,49 @@ async def score_attempt(
 
         return ret.attempt_scored_model
 
-    async def get_static_file(self, path: str) -> PackageFileStream:
-        raise NotImplementedError  # TODO: Implement get_static_file.
+    def _get_static_file_sync(self, path: str, manifest: Manifest) -> PackageFileData:
+        path = path.lstrip("/")
+
+        try:
+            manifest_entry = manifest.static_files[path]
+        except KeyError as e:
+            raise FileNotFoundError(path) from e
+
+        dist_path = DIST_DIR + "/" + path.lstrip("/")
+
+        reader: Callable[[], bytes]
+        if isinstance(self.package, ZipPackageLocation):
+            with ZipFile(self.package.path) as zip_file:
+                try:
+                    zipinfo = zip_file.getinfo(dist_path)
+                except KeyError as e:
+                    raise FileNotFoundError(path) from e
+
+                real_size = zipinfo.file_size
+                reader = partial(zip_file.read, dist_path)
+        elif isinstance(self.package, DirPackageLocation):
+            full_path: Path = self.package.path / dist_path
+            real_size = full_path.stat().st_size
+            reader = full_path.read_bytes
+        elif isinstance(self.package, FunctionPackageLocation):
+            msg = f"'{path}'. Function-based packages don't serve static files."
+            raise FileNotFoundError(msg)
+        else:
+            raise TypeError(type(self.package).__name__)
+
+        if manifest_entry.size != real_size:
+            msg = (
+                f"Static file '{path}' has different file size on disk ('{real_size}') than in manifest "
+                f"('{manifest_entry.size}')"
+            )
+            raise RuntimeError(msg)
```

Comment:
Wie reagiert aiohttp auf diese Exception? Ich glaube ein `HTTPInternalServerError` könnte besser sein und zusätzlich ein Log-Eintrag. Fehler, die durch die Pakete verursacht werden, sollten wir maximal als warning betrachten, denke ich.

Reply:
Da das eh bald im Worker landen wird, war mir schönes Error-Handling weniger wichtig als Geschwindigkeit.

Sauber würde ich eine eigene Exception dafür definieren, hier eine aiohttp-Exception zu werfen würde Abstraktionsebenen vermischen, das müsste dann in der Route passieren.

Loggen ist sinnvoll :+1:, ich würde `INFO` nehmen. `WARN` sollte denke ich immer heißen, dass sich das demnächst mal jemand von uns anschauen und vermutlich eingreifen muss.